tests/unit/test_provider.py

```diff
@@ -6,68 +6,58 @@
 import re
 import unittest
 from collections import namedtuple
+from pathlib import Path
 
 import ops.testing
-from charms.rolling_ops.v0.rollingops import RollingOpsManager
-from ops.charm import CharmBase, RelationBrokenEvent
+import yaml
+from ops.charm import RelationBrokenEvent
 from ops.testing import Harness
 
-from cluster import ZooKeeperCluster
-from provider import ZooKeeperProvider
+from charm import ZooKeeperK8sCharm
+from literals import CHARM_KEY, PEER, REL_NAME
 
 ops.testing.SIMULATE_CAN_CONNECT = True
 
 logger = logging.getLogger(__name__)
 
-METADATA = """
-    name: zookeeper
-    peers:
-        cluster:
-            interface: cluster
-        restart:
-            interface: rolling_op
-    provides:
-        zookeeper:
-            interface: zookeeper
-"""
+METADATA = str(yaml.safe_load(Path("./metadata.yaml").read_text()))
+CONFIG = str(yaml.safe_load(Path("./config.yaml").read_text()))
+ACTIONS = str(yaml.safe_load(Path("./actions.yaml").read_text()))
 
 CustomRelation = namedtuple("Relation", ["id"])
 
 
-class DummyZooKeeperCharm(CharmBase):
-    def __init__(self, *args):
-        super().__init__(*args)
-        self.cluster = ZooKeeperCluster(self)
-        self.client_relation = ZooKeeperProvider(self)
-        self.restart = RollingOpsManager(self, relation="restart", callback=lambda x: x)
-
-
 class TestProvider(unittest.TestCase):
     def setUp(self):
-        self.harness = Harness(DummyZooKeeperCharm, meta=METADATA)
+        self.harness = Harness(ZooKeeperK8sCharm, meta=METADATA, config=CONFIG, actions=ACTIONS)
         self.addCleanup(self.harness.cleanup)
-        self.harness.add_relation("zookeeper", "application")
-        self.harness.begin_with_initial_hooks()
+        self.harness.add_relation(REL_NAME, "application")
+        self.harness.add_relation(PEER, CHARM_KEY)
+        self.harness.begin()
 
     @property
     def provider(self):
-        return self.harness.charm.client_relation
+        return self.harness.charm.provider
 
     def test_relation_config_new_relation_no_chroot(self):
-        config = self.harness.charm.client_relation.relation_config(
+        config = self.harness.charm.provider.relation_config(
             relation=self.provider.client_relations[0]
         )
         self.assertIsNone(config)
 
     def test_relation_config_new_relation(self):
+
         self.harness.update_relation_data(
             self.provider.client_relations[0].id, "application", {"chroot": "app"}
         )
         self.harness.update_relation_data(
-            self.provider.app_relation.id, "zookeeper", {"relation-0": "password"}
+            self.provider.app_relation.id, CHARM_KEY, {"relation-0": "password"}
         )
 
-        config = self.harness.charm.client_relation.relation_config(
+        logger.info(self.provider.client_relations[0].data)
+        logger.info(self.provider.app_relation.data)
```

Comment:
Are these logs only related to debugging during test development and should they be removed?

Reply:
Very good catch, thank you!!